docs/JLBP-16.md

```diff
@@ -1,27 +1,37 @@
-# [JLBP-16] Ensure upper version alignment of dependencies for consumers
+# [JLBP-16] Ensure upper version alignment of dependencies
 
 For multi-module projects, this best practice assumes you have already applied
 [JLBP-15](JLBP-15.md), so that your project has a BOM.
 
-Upper version alignment is defined in the [glossary](glossary.md).
+Upper version alignment means that when more than one version of a dependency
+is present in a module's dependency tree, the build system selects
+the highest version. Upper version alignment ensures that when packages are
+upgraded to higher versions which don’t make any breaking changes,
+there are no new linkage conflicts.
```

Comment:
In my view, upper version alignment does not ensure no new linkage conflicts. I would replace "Upper version alignment ensures ..." with the below:

> In general, selecting the highest version of a library is a good choice. The version has more security fixes and is more compatible with other libraries than the lower versions of it.